dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -445,17 +445,17 @@ Sedp::init(const RepoId& guid,
 #endif
 
   if (spdp_.available_builtin_endpoints() & DISC_BUILTIN_ENDPOINT_PARTICIPANT_ANNOUNCER) {
-    participant_message_writer_->enable_transport_using_config(reliable, durable, transport_cfg_);
+    participant_message_writer_->enable_transport_using_config(reliable, nondurable, transport_cfg_);
   }
-  participant_message_reader_->enable_transport_using_config(reliable, durable, transport_cfg_);
+  participant_message_reader_->enable_transport_using_config(reliable, nondurable, transport_cfg_);
 
 #ifdef OPENDDS_SECURITY
   participant_message_secure_writer_->set_crypto_handles(spdp_.crypto_handle());
   participant_message_secure_reader_->set_crypto_handles(spdp_.crypto_handle());
   if (spdp_.available_builtin_endpoints() & DDS::Security::BUILTIN_PARTICIPANT_MESSAGE_SECURE_WRITER) {
-    participant_message_secure_writer_->enable_transport_using_config(reliable, durable, transport_cfg_);
```

Comment:
And this seems to violate 7.4.4.1 of the Security spec (see last bullet in particular, stating that it's the reason the volatile readers / writers were necessary).

Reply:
I'll defer to @mitza-oci

Reply:
7.4.4.1 is non-normative.  It provides background on why DCPSParticipantVolatileMessageSecure exists in the spec.